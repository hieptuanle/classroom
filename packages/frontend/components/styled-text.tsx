import type { TextProps } from "./themed";

import { Text } from "./themed";

export function MonoText(props: TextProps) {
  const { style, ...textProps } = props;
  return <Text {...textProps} style={[style, { fontFamily: "SpaceMono" }]} />;
}
