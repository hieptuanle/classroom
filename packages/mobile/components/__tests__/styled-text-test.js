import { render } from "@testing-library/react-native";

import { MonoText } from "../styled-text";

it("renders correctly", () => {
  const { getByText } = render(<MonoText>Snapshot test!</MonoText>);

  // Test that the text is rendered
  expect(getByText("Snapshot test!")).toBeTruthy();
});

it("matches snapshot", () => {
  const tree = render(<MonoText>Snapshot test!</MonoText>).toJSON();
  expect(tree).toMatchSnapshot();
});
