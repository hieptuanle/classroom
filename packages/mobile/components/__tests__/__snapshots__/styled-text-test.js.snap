// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Text
  style={
    [
      {
        "color": "#000",
      },
      [
        undefined,
        {
          "fontFamily": "SpaceMono",
        },
      ],
    ]
  }
>
  Snapshot test!
</Text>
`;
